src/ResourceManagement/Compute/EncryptionExtensionIdentifier.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.Compute.Fluent.Models
+{
+    /// <summary>
+    /// An internal type expose utility methods around encryption extension identifier.
+    /// </summary>
+    public partial class EncryptionExtensionIdentifier
+    {
+        private const string ENCRYPTION_EXTENSION_PUBLISHER = "Microsoft.Azure.Security";
+
+        private const string LINUX_LEGACY_ENCRYPTION_EXTENSION_VERSION = "0.1";
+        private const string LINUX_NOAAD_ENCRYPTION_EXTENSION_VERSION = "1.1";
+
+        private const string WINDOWS_LEGACY_ENCRYPTION_EXTENSION_VERSION = "1.1";
+        private const string WINDOWS_NOAAD_ENCRYPTION_EXTENSION_VERSION = "2.2";
+
+        private const string LINUX_ENCRYPTION_TYPE_NAME = "AzureDiskEncryptionForLinux";
+        private const string WINDOWS_ENCRYPTION_TYPE_NAME = "AzureDiskEncryption";
+
+        /// <summary>
+        /// Return encryption extension publisher name.
+        /// </summary>
+        /// <returns>encryption extension publisher name</returns>
+        internal static string PublisherName()
```

Comment:
For method naming convention, you either provide access method such as `getPublisherName()` or provide C# property. Similar for other methods in this class.

Reply:
Done